File: backend/src/main/java/com/it/finance/personal_finance_be/dto/MovementDto.java
package com.it.finance.personal_finance_be.dto;

import com.it.finance.personal_finance_be.framework.PfDto;

import java.math.BigDecimal;
import java.time.LocalDateTime;

public class MovementDto extends PfDto {

    private LocalDateTime dt;

    private EconomicAccountDto economicAccount;

    private String note;

    private PatrimonialFundDto patrimonialFund;

    private BigDecimal amount;

    private Long blockId;

    public LocalDateTime getDt() {
        return dt;
    }

    public void setDt(LocalDateTime dt) {
        this.dt = dt;
    }

    public EconomicAccountDto getEconomicAccount() {
        return economicAccount;
    }

    public void setEconomicAccount(EconomicAccountDto economicAccount) {
        this.economicAccount = economicAccount;
    }

    public String getNote() {
        return note;
    }

    public void setNote(String note) {
        this.note = note;
    }

    public PatrimonialFundDto getPatrimonialFund() {
        return patrimonialFund;
    }

    public void setPatrimonialFund(PatrimonialFundDto patrimonialFund) {
        this.patrimonialFund = patrimonialFund;
    }

    public BigDecimal getAmount() {
        return amount;
    }

    public void setAmount(BigDecimal amount) {
        this.amount = amount;
    }

    public Long getBlockId() {
        return blockId;
    }

    public void setBlockId(Long blockId) {
        this.blockId = blockId;
    }
}

File: frontend/src/dto/finance/MovementDto.ts
import PfDto from "@/dto/framework/PfDto.ts";
import type EconomicAccountDto from "@/dto/finance/EconomicAccountDto";
import type PatrimonialFundDto from "@/dto/finance/PatrimonialFundDto";

export default class MovementDto extends PfDto {

    dt:Date | null;
    economicAccount: EconomicAccountDto | null = null;
    note:string | null;
    patrimonialFund: PatrimonialFundDto | null = null;
    amount: number | null;
    blockId: number | null;

}